src/main/java/yanagishima/interceptor/DatasourceInterceptor.java

```diff
@@ -0,0 +1,48 @@
+package yanagishima.interceptor;
+
+import static yanagishima.util.AccessControlUtil.sendForbiddenError;
+import static yanagishima.util.AccessControlUtil.validateDatasource;
+
+import java.lang.reflect.Method;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import org.springframework.web.servlet.resource.ResourceHttpRequestHandler;
+
+import lombok.RequiredArgsConstructor;
+import yanagishima.annotation.DatasourceAuth;
+import yanagishima.config.YanagishimaConfig;
+
+@Component
+@RequiredArgsConstructor
+public class DatasourceInterceptor extends HandlerInterceptorAdapter {
+  private final YanagishimaConfig config;
+
+  @Override
+  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+    if (handler instanceof ResourceHttpRequestHandler) {
+      return true;
+    }
+
+    if (!(handler instanceof HandlerMethod)) {
+      return true;
+    }
+
+    Method method = ((HandlerMethod) handler).getMethod();
+    DatasourceAuth annotation = AnnotationUtils.findAnnotation(method, DatasourceAuth.class);
+    if (annotation == null) {
+      return true;
+    }
+
+    if (config.isCheckDatasource() && !validateDatasource(request, request.getParameter("datasource"))) {
+      sendForbiddenError(response);
+      return false;
+    }
+    return true;
```

Comment:
I have some questions:
Is it because we need `HttpServletRequest/Response`, so the implementation of DatasourceAuth is placed here?
When we have several other customs annotations, the logic may be a little complex to manage them like this way. 
Did you consider other ways to implement (just curious)

Reply:
> Is it because we need HttpServletRequest/Response, so the implementation of DatasourceAuth is placed here?

Yes. I don't think codes will be complex even if we add new annotations later. Though I considered other approach, choosing this for simplicity.

Reply:
I think this approach is not so complex.

Reply:
Some explanations: my original consideration is that 
suppose we have annotations `@A` and `@DatasourceAuth`  if a method is annotated `@A` only and we implement annotation in this way
we need to move the check logic into the block
```
 if (annotation == null) {
      return true;
}
```
when the annotation increasing, the nested checked may get deeper, so that I wonder whether there is another way to achieve it. Of course, if the new annotation does not require `HttpServletRequest/Response`, the logic is not needed to be implemented here.